.github/workflows/ci.yml

```diff
@@ -87,6 +87,48 @@ jobs:
         if: steps.cache.outputs.cache-hit != 'true' && matrix.cache == 'true'
         run: rm -rf ~/.m2/repository/io/trino/trino-*
 
+  reproducible-build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # checkout all commits, as the build result depends on `git describe` equivalent
+          ref: |
+            ${{ github.event_name == 'repository_dispatch' &&
+                github.event.client_payload.pull_request.head.sha == github.event.client_payload.slash_command.args.named.sha &&
+                format('refs/pull/{0}/head', github.event.client_payload.pull_request.number) || '' }}
+      - uses: ./.github/actions/setup
+        with:
+          cache: 'false'
+          cleanup-node: true
+      - name: First build to a local repository
+        # All dependencies are downloaded to and artifacts generated to build_1 directory, .m2 isn't used
```

Comment:
- .m2 -> ~/.m2/repository  (in particular, i assume other files from ~/.m2 may still be used)
-  isn't used ->  isn't used because ...